addons/misra.py

```diff
@@ -3333,7 +3333,9 @@ def misra_17_7(self, data):
                 continue
             if token.str != '(' or token.astParent:
                 continue
-            if not token.previous.isName or token.previous.varId:
+            if not token.astOperand1 or not token.astOperand1.isName:
+                continue
+            if token.astOperand1.varId and None == get_function_pointer_type(token.astOperand1.variable.typeStartToken):
```

Comment:
I believe it is more pythonic to compare to None using `is`.

Reply:
I agree. Fixed it up :)